op-geth-proxy/geth-proxy.go

```diff
@@ -0,0 +1,112 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"flag"
+	"io"
+	"log"
+	"net/http"
+	"net/http/httputil"
+	"net/url"
+	"os"
+
+	"github.com/peterbourgon/ff/v3"
+)
+
+// Environment variables beginning with this prefix can be used to instantiate command line flags
+const ENV_PREFIX = "SEQUENCER_PROXY"
+
+var (
+	fs            = flag.NewFlagSet("proxy", flag.ContinueOnError)
+	fromAddr      = fs.String("listen-addr", "127.0.0.1:9090", "proxy's listening adress")
+	sequencerAddr = fs.String("seq-addr", "http://127.0.0.1:50000", "address of espresso sequencer")
+	gethAddr      = fs.String("geth-addr", "http://127.0.0.1:8545", "address of the op-geth node")
+	vm_id         = fs.Int("vm-id", 1, "VM ID of the OP rollup instance")
+)
+
+type Transaction struct {
+	Vm      int   `json:"vm"`
+	Payload []int `json:"payload"`
+}
+
+type rpcMessage struct {
+	Params json.RawMessage `json:"params,omitempty"`
+	Method string          `json:"method,omitempty"`
+}
+
+func ForwardToSequencer(message rpcMessage) {
+	// json.RawMessage is a []byte array, which is marshalled
+	// As a base64-encoded string. Our sequencer API expects a JSON array.
+	payload := make([]int, len(message.Params))
+	for i := range payload {
+		payload[i] = int(message.Params[i])
+	}
```

Comment:
I'm not sure this is quite right. `json.RawMessage` is just serialized JSON object. Since in this case it corresponds to the "params" field of the JSON-RPC request, it will be a JSON array object containing, for `eth_sendRawTransaction`, one parameter. That parameter in turn is a JSON string which is a base64 encoding of the signed transaction. So I think we need to
* change the type of `Params` to `[]json.RawMessage`, so that in `ServeHTTP` we automatically unmarshal the outer JSON array (since all RPC requests have this form)
* decode the parameter from a base64 string to a byte array

At least that's my understanding, lmk if it sounds wrong

Reply:
Ah yes you're right - I still think we need to do the annoying []int to []byte conversion in the end for the reasons I explained above.

Reply:
json.RawMessage is just a wrapper for []byte, so the first element of Params after this change should be the byte array we need to forward

Reply:
I don't think so, because IIUC the `[]byte` inside `RawMessage` is really a JSON object represented as an `ASCII` string. The first parameter to the `eth_sendRawTransaction` endpoint is, I *think*, a string that is hex or base64 encoded, representing the bytes of the transaction. So the first element of Params will be a byte array like `['"', '0', 'x', ...more hex chars..., '"']` or something. We actually need to call a hex or base64 decoder to convert that into the underlying bytes

Reply:
https://docs.infura.io/networks/ethereum/json-rpc-methods/eth_sendrawtransaction

It looks like the encoding is hex with the `0x` prefix